### PR TITLE
[Vectra_Detect] Initial release for the Vectra Detect

### PR DESCRIPTION
**Pull request by piyush-elastic**
## What does this PR do?

- Updated data collection logic for the log data stream.
- Updated the ingest pipeline for the log data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Updated test for pipeline for the log data stream.
- Updated system test cases for the log data stream.

## Integration release checklist
This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes
- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target is documented
- [x] Supported operating systems are documented (if applicable)
- [x] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to: ^8.3.0

### New Package
- [x] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes
- [x] Dashboards exists
- [x] Screenshots added or updated
- [x] Datastream filters added to visualizations

### Log dataset changes
- [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [x] Generated output for at least 1 log file exists
- [x] Sample event (sample_event.json) exists

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/vectra_detect directory.
- Run the following command to run tests. 
> elastic-package test 

## Screenshots
Integration
Overview
pipeline
static
system
asset

### CODE REVIEW

**Review by bhapas — COMMENTED** [review]
Looks good to me.. But I would let one of our team members have a closer look at pipelines

File: packages/vectra_detect/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,93 @@
+Elastic License 2.0
```
Comment (kgeller) [inline packages/vectra_detect/LICENSE.txt:1]: Why do we include the license here? I don't recall seeing that on other packages

Comment (piyush-elastic) [reply packages/vectra_detect/LICENSE.txt:1]: @kgeller , as per new version of elastic-package license file is getting created at package creation level reference - https://github.com/elastic/elastic-package/pull/898 and before that we had been using old elastic-package version.

File: packages/vectra_detect/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      copy_from: log.syslog.hostname
+      ignore_empty_value: true
+  - set:
+      field: vectra_detect.log.syslog
+      copy_from: log.syslog
+      ignore_empty_value: true
+  - dissect:
+      field: event.original
+      pattern: '%{vectra_detect.log.event_type} -: %{_tmp.message}'
+      ignore_failure: true
```
Comment (kcreddy) [inline packages/vectra_detect/data_stream/log/elasticsearch/ingest_pipeline/default.yml:37]: I dont think you should `ignore_failure` here. This seems to be important parsing that leads further processing of the pipeline. Can you add `on_failure` like below `trim`?

File: packages/vectra_detect/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      field: json.version
+      target_field: vectra_detect.log.version
+      ignore_missing: true
+  - set:
+      field: observer.version
+      copy_from: vectra_detect.log.version
+      ignore_empty_value: true
+  - pipeline:
+      name: '{{ IngestPipeline "pipeline-host-scoring" }}'
+      if: ctx.vectra_detect?.log?.event_type == 'vectra_json_v2' && ctx.json?.category == 'HOST SCORING'
```
Comment (kcreddy) [inline packages/vectra_detect/data_stream/log/elasticsearch/ingest_pipeline/default.yml:108]: When this condition is met, the event goes through both `pipeline-host-scoring` and `pipeline-host-detection` pipelines. Is this intended?

File: packages/vectra_detect/data_stream/log/_dev/test/pipeline/test-account-detection.log-expected.json [hunk 4]
```diff
@@ -0,0 +1,797 @@
...
+            "related": {
+                "hosts": [
+                    "x29-1-13.sc.tvec"
+                ],
+                "ip": [
+                    "89.160.20.128",
+                    "0.0.0.0"
+                ],
+                "user": [
+                    "svc-x45h8@corp.example.com"
```
Comment (kcreddy) [inline packages/vectra_detect/data_stream/log/_dev/test/pipeline/test-account-detection.log-expected.json:54]: All `account_uid` values can come inside `related.user`. Same for all other pipelines

File: packages/vectra_detect/data_stream/log/_dev/test/pipeline/test-account-detection.log-expected.json [hunk 5]
```diff
@@ -0,0 +1,797 @@
...
+            "url": {
+                "domain": "x29-1-13.sc.tvec",
+                "original": "https://x29-1-13.sc.tvec/detections/85?detail_id=378",
+                "path": "/detections/85",
+                "query": "detail_id=378",
+                "scheme": "https"
+            },
+            "user": {
+                "target": {
+                    "name": "svc-x45h8@corp.example.com"
```
Comment (kcreddy) [inline packages/vectra_detect/data_stream/log/_dev/test/pipeline/test-account-detection.log-expected.json:73]: If this is email, it should also be present in `user.email` and `user.id` fields. You can also capture `user.domain`

Comment (piyush-elastic) [reply packages/vectra_detect/data_stream/log/_dev/test/pipeline/test-account-detection.log-expected.json:73]: @kcreddy ,As per [documentation](https://support.vectra.ai/s/relatedlist/ka06S000000olopQAA/AttachedContentDocuments
) page # 39, description of field `account_uid` , is _'The account name'_ , so I don't think we can always consider it as an email id. Let me know your thoughts on this.

File: packages/vectra_detect/data_stream/log/_dev/test/pipeline/test-account-detection.log-expected.json (hunk 4 above)
Comment (kcreddy) [reply packages/vectra_detect/data_stream/log/_dev/test/pipeline/test-account-detection.log-expected.json:56]: Hey @piyush-elastic 
I didnt meant to split the `account_uid ` field. 
What I meant was that all values of `account_uid` inside array `account_info` or another array should be copied into `related.user`.

File: packages/vectra_detect/data_stream/log/elasticsearch/ingest_pipeline/pipeline-account-detection.yml [hunk 6]
```diff
@@ -65,6 +65,36 @@ processors:
...
+      copy_from: vectra_detect.log.account.domain
+      ignore_empty_value: true
+  - registered_domain:
+      field: user.domain
+      target_field: vectra_detect.log.user
+      ignore_missing: true
+  - set:
+      field: user.id
+      copy_from: vectra_detect.log.account.user_id
+      ignore_empty_value: true
```
Comment (kcreddy) [inline packages/vectra_detect/data_stream/log/elasticsearch/ingest_pipeline/pipeline-account-detection.yml:87]: What I meant was after `dissect`:
1. If both `user_id` and `domain` are not null, then populate `user.id` and `user.email` with entire value of `account.uid`.
2. If  `domain` is not null, populate `user.domain` -> this is done. Thanks!

**Review by kcreddy — APPROVED** [review]
LGTM

**Pull request merged**